app/models/user.rb

```diff
@@ -240,7 +240,7 @@ def claim_wca_id_validations
         errors.add(:unconfirmed_wca_id, I18n.t('users.errors.already_have_id', wca_id: wca_id))
       end
 
-      if delegate_id_to_handle_wca_id_claim.present? && !delegate_to_handle_wca_id_claim
+      if delegate_id_to_handle_wca_id_claim.present? && !User.find_by(id: delegate_id_to_handle_wca_id_claim).any_kind_of_delegate?
```

Comment:
Why are you changing this line? The `belongs_to` above should still work...

Reply:
One of the test-case which was failing because of this was "delegate_id_to_handle_wca_id_claim must be a delegate". The `delegate_to_handle_wca_id_claim` (the `belongs_to`) can be a non-delegate. The association-way of checking might be bit complicated, hence chose this way of checking if the delegate_id_to_handle_wca_id_claim is a delegate or not.

Reply:
In that case, you should probably rename the above `belongs_to` thingy to `user_to_handle_wca_id_claim` and change this call accordingly to `user_to_handle_wca_id_claim.any_kind_of_delegate?`. I'd like to keep manual "boring" `find_by` lookups (i.e. `find` that uses only `id` and nothing else) to an absolute minimum.

Reply:
In that case, may I offer a compromise: Just keep the name as-is but still use the association here, despite the fact that the name might be confusing?